examples/twittermap/web/public/javascripts/map/pointmap.js

```diff
@@ -253,20 +256,21 @@ angular.module('cloudberry.map')
               fillColor : '#b8e3ff',
               fillOpacity : 1.0
             }).addTo($scope.map);
-            //(3) Send request to twitter.com for the oembed json tweet content.
-            var url = "https://api.twitter.com/1/statuses/oembed.json?callback=JSON_CALLBACK&id=" + $scope.pointIDs[i];
-            $http.jsonp(url).success(function (data) {
-              var tweetContent = translateOembedTweet(data);
-              $scope.popUpTweet = L.popup({maxWidth:300, minWidth:300, maxHight:300});
-              $scope.popUpTweet.setContent(tweetContent);
-              $scope.currentMarker.bindPopup($scope.popUpTweet).openPopup();
-            }).
-            error(function() {
-              var tweetContent = "Sorry! It seems the tweet with that ID has been deleted by the author.@_@";
+
+            //send the query to cloudberry.
+            var passID = "" + $scope.pointIDs[i];
+            cloudberry.pinMapOneTweetQuery(passID);
+
+            //receives the result and update content of tweet.
```

Comment:
For this part, I think the logic is not correct, please pay attention to the line 330-335 in this file.
The $watch function should accept 2 parameters: first is a function who returns the variable that we need to watch for its update, second is a function that will be called when the first variable is updated. Try to mimic that logic.